lib/trino-hdfs/src/main/java/io/trino/filesystem/hdfs/cache/CachingFileSystemConfig.java

```diff
@@ -0,0 +1,69 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.trino.filesystem.hdfs.cache;
+
+import io.airlift.configuration.Config;
+import io.airlift.configuration.ConfigDescription;
+import io.airlift.units.DataSize;
+
+import javax.annotation.Nullable;
+
+import static io.airlift.units.DataSize.Unit.GIGABYTE;
+
+public class CachingFileSystemConfig
+{
+    private boolean cacheEnabled;
+    private String baseDirectory;
+    private DataSize maxCacheSize = DataSize.of(2, GIGABYTE);
```

Comment:
This is not a great default value as it will result in data going into and out of cache very often.
Ideally we want to configure this as a percentage of available space like `spiller-max-used-space-threshold`

Reply:
Unfortunately we're limited to what Alluxio offers here, as this maps to the `alluxio.user.client.cache.size` parameter which only accepts actual sizes rather than percentages.

I've set both this and the path to their Alluxio defaults (`512MB` and `/tmp/alluxio_cache` respectively) based on the assumption that each Trino user that enables the cache will tweak these values based on their installation, as they'll have to provision cache disks etc anyway. I'm open to set these to different values though; as Trino doesn't really have to follow these here 👍 What do you have in mind?

Reply:
The default cache location should be empty and the user should be forced to provide a location whenever this cache is enabled (See RubixEnabledConfig and RubixConfig#cacheLocation for example).
We should use something like 70-80% of total space on the provided cache location as a default cache size parameter to alluxio (See some of the code in FileSingleStreamSpillerFactory)

Reply:
> The default cache location should be empty and the user should be forced to provide a location whenever this cache is enabled (See RubixEnabledConfig and RubixConfig#cacheLocation for example).

Makes sense 👍 will implement this.

> We should use something like 70-80% of total space on the provided cache location as a default cache size parameter to alluxio (See some of the code in FileSingleStreamSpillerFactory)

Rather than providing an arbitrary default value here, shouldn't we force the user to provide a value, for consistency with what we do for the cache directories?

Reply:
> Rather than providing an arbitrary default value here, shouldn't we force the user to provide a value, for consistency with what we do for the cache directories?

It's not arbitrary, when a cache directory is provided by the user, it's reasonable to assume that they want to use most of it for cached data